rtribbia/template/memory-gui.js

```diff
@@ -0,0 +1,122 @@
+var MemoryGUI = (function () {
+
+
+	function GuiCtor(container,game) {
+		var cont;
+		if (container instanceof HTMLElement) {
+			cont = container;
+		} else if ((typeof container) == "string") {
+			cont = document.getElementById(container);
+		} else { return {}};
+
+		// public instance methods:
+		function reset() {
+			//...
+			while (cont.firstChild) { //from stack overflow- clear DOM
+    			cont.removeChild(cont.firstChild);
+			}
+			game.reset(); //reset game state
+			renderBoard(); //re-render board
+			alert('board has been reset!');
+		}
+		function show(id) { //only need one parameter because board[id] and td =id  are the same in the way I ended up implementing it
+			//...
+			var target = document.getElementById(id);
+			target.innerHTML = "";
+			var img = document.createElement('img');
+	        src = 'images/SVG-cards-1.3/' + game.getImg(id) + '.svg';
+	        img.src = src;
+	        img.setAttribute('width',65);
+	        target.innerHTML = ""
+	        target.appendChild(img);
+		}
+
+		function removeSoon(whereArr) {
+			//...
+			window.setTimeout(function() {
+				whereArr.forEach(function(x){
+					document.getElementById(x).classList.toggle('invisible');
+				});
+			}, 1000);
+		}
+		function hideSoon(whereArr) {
+			//...
+			window.setTimeout(function() {
+				whereArr.forEach(function(x){
+					document.getElementById(x).innerHTML = "";
+					document.getElementById(x).appendChild(renderCard_back());
+				});
+			}, 1000);			
+		}
+		function lift(evt) {
+			var result = game.lift(this.id); //[match?, where, last faceup card (if any)]
+			if (result[0] == 'match') {
+				console.log('Match! - Making these invisible!\nnew:' + this.id + '\nold:' + result[2]);
```

Comment:
This is where you could insert a class definition for the two matched cards. Essentially, as soon as the cards are matched, you could add a class with related CSS styles that would change the appearance of the cards in some way. Then, after the delay from `removeSoon`, you would want to be sure that the cards were still invisible.